### DISCUSSION TITLE
Consider construct port for  fitting potential including the second-order, third-order, and fourth-order derivatives of force?

### DISCUSSION THREAD

**Discussion by davy12344** [deepmodeling/deepmd-kit#4475] [Ideas]
Is it possible to have a port that incorporates the second-order, third-order, and fourth-order derivatives of force (intentionally constructing such datasets) into the fitting of potential functions? 

I've found that considering only the fitting of energy, force, and virial, when calculating the third-order force constant and fourth-order force constant for BTE to compute thermal conductivity, there might be errors in the calculated force, and these errors can accumulate, leading to inaccurate thermal conductivity calculations. 

If we consider the second-order, third-order, and fourth-order derivatives of force in the fitting of potential functions, could it aid in such BTE calculations? Is it possible to complete such a fit by do something with DEEPMD? If so, what would I generally need to do? 

Looking forward to receive your advice.

**Comment by njzjz** [comment]
#4169 seems to be aimed at the second-order derivatives.
